actions/watch.ts

```diff
@@ -83,101 +97,123 @@ export const checkForAndPlaceOrder: ActionFn = async (
   const blockEvent = event as BlockEvent;
   const registry = await Registry.load(context, blockEvent.network);
   const chainContext = await ChainContext.create(context, blockEvent.network);
-
-  console.log(`Processing block ${blockEvent.blockNumber}...`);
+  const { network } = blockEvent;
 
   // enumerate all the owners
   for (const [owner, conditionalOrders] of registry.ownerOrders.entries()) {
-    console.log(`Checking ${owner}...`);
-
+    const deletedOrders = [];
     // enumerate all the `ConditionalOrder`s for a given owner
     for (const conditionalOrder of conditionalOrders) {
-      console.log(`Checking params ${conditionalOrder.params}...`);
+      // console.log(`Checking params ${conditionalOrder.params}...`);
       const contract = ComposableCoW__factory.connect(
         conditionalOrder.composableCow,
         chainContext.provider
       );
-      try {
-        const { order, signature } =
-          await contract.callStatic.getTradeableOrderWithSignature(
-            owner,
-            conditionalOrder.params,
-            "0x",
-            conditionalOrder.proof ? conditionalOrder.proof.path : []
-          );
 
-        const orderToSubmit: Order = {
-          ...order,
-          kind: kindToString(order.kind),
-          sellTokenBalance: balanceToString(order.sellTokenBalance),
-          buyTokenBalance: balanceToString(order.buyTokenBalance),
-        };
+      const { deleteConditionalOrder } = await _checkForAndPlaceOrder(
+        owner,
+        network,
+        conditionalOrder,
+        contract,
+        chainContext
+      );
 
-        // calculate the orderUid
-        const orderUid = computeOrderUid(
-          {
-            name: "Gnosis Protocol",
-            version: "v2",
-            chainId: blockEvent.network,
-            verifyingContract: GPV2SETTLEMENT,
-          },
-          {
-            ...orderToSubmit,
-            receiver:
-              orderToSubmit.receiver === ethers.constants.AddressZero
-                ? undefined
-                : orderToSubmit.receiver,
-          },
-          owner
-        );
+      if (deleteConditionalOrder) {
+        deletedOrders.push(conditionalOrder);
```

Comment:
I moved the deletion out of the loop. I would think is not a good idea in general to mutate the collection you are iterating while you are interesting on it

Reply:
Great catch, makes sense!